### PR TITLE
policy: Add config for enabling Cilium NetworkPolicy

### PR DESCRIPTION
**Pull request by dlapcevic**
Related to modularizing network policies: https://github.com/cilium/cilium/issues/33360

Signed-off-by: Dorde Lapcevic <[dordel@google.com](mailto:dordel@google.com)>

### CODE REVIEW

**Review by sypakine — COMMENTED** [review]
- Should CCNP have separate enablement?
- Should this short-circuit validation? operator/pkg/networkpolicy/cell.go
- Should this disable the operator watchers? operator/cmd/root.go

**Comment by dlapcevic** [comment]
Great questions @sypakine!

> * Should CCNP have separate enablement?

It would be good to differentiate CNP and CCNP. I propose to do it as a follow up. For now one flag will stand for both CNP and CCNP.

> * Should this short-circuit validation? operator/pkg/networkpolicy/cell.go

Yes, it should. I added it, and also for cilium-operator to read the same flag.

> * Should this disable the operator watchers? operator/cmd/root.go

Yes, it should. I added it. It reuses the same cilium-operator from the previous point.

File: pkg/option/config.go [hunk 1]
```diff
@@ -3588,6 +3600,9 @@ func (c *DaemonConfig) Populate(vp *viper.Viper) {
 	c.EnableNodeSelectorLabels = vp.GetBool(EnableNodeSelectorLabels)
 	c.NodeLabels = vp.GetStringSlice(NodeLabels)
 
+	// To support Cilium NetworkPolicy
```
Comment (sypakine) [inline pkg/option/config.go]: nit: redundant

File: daemon/cmd/daemon_main.go [hunk 2]
```diff
@@ -1132,6 +1132,10 @@ func InitGlobalFlags(cmd *cobra.Command, vp *viper.Viper) {
 	flags.MarkHidden(option.EnableK8sNetworkPolicy)
 	option.BindEnv(vp, option.EnableK8sNetworkPolicy)
 
+	flags.Bool(option.EnableCiliumNetworkPolicy, defaults.EnableCiliumNetworkPolicy, "Enable support for Cilium NetworkPolicy")
```
Comment (sypakine) [inline daemon/cmd/daemon_main.go]: I don't think you merged the changes to update the flag/comments to clarify this enables both CNP and CCNP?

File: operator/pkg/networkpolicy/cell.go [hunk 3]
```diff
@@ -72,6 +72,11 @@ func registerPolicyValidator(params PolicyParams) {
 		return
 	}
 
+	if !option.Config.EnableCiliumNetworkPolicy {
+		params.Logger.Debug("CNP / CCNP validator doesn't run when CNP / CCNP are disabled")
```
Comment (ovidiutirla) [inline operator/pkg/networkpolicy/cell.go]: nit: should we make this one info?
* Include the config key and its value as part of the logging.

File: pkg/option/config.go [hunk 4]
```diff
@@ -2724,6 +2731,11 @@ func (c *DaemonConfig) K8sNetworkPolicyEnabled() bool {
 	return c.EnableK8sNetworkPolicy
 }
 
+// CiliumNetworkPolicyEnabled returns true if cilium agent needs to support Cilium NetworkPolicy, false otherwise.
+func (c *DaemonConfig) CiliumNetworkPolicyEnabled() bool {
```
Comment (sypakine) [inline pkg/option/config.go]: Unused?

**Review by sypakine — APPROVED** [review]
LGTM with nit about the flag/comment lacking clarity that this applies to both CNP and CCNP.

File: pkg/option/config.go (hunk 1 above)
Comment (dlapcevic) [reply pkg/option/config.go]: Done.

File: daemon/cmd/daemon_main.go (hunk 2 above)
Comment (dlapcevic) [reply daemon/cmd/daemon_main.go]: Done.

File: pkg/option/config.go (hunk 4 above)
Comment (dlapcevic) [reply pkg/option/config.go]: Removed.

**Review by ovidiutirla — APPROVED** [review]
🚀

**Review by joamaki — APPROVED** [review]

File: operator/pkg/networkpolicy/cell.go (hunk 3 above)
Comment (dlapcevic) [reply operator/pkg/networkpolicy/cell.go]: Done

**Review by asauber — APPROVED** [review]

**Comment by doniacld** [comment]
Adding @squeed to reviewers to double check for sig-policy group.

**Review by doniacld — APPROVED** [review]

**Pull request merged**